### PR TITLE
Melee hit tables updated to classic values

### PR DESCRIPTION
**Pull request by mako-taco**
Updated the melee roll calculations for dodge/miss/crit/parry based on the blue post  here: https://us.forums.blizzard.com/en/wow/t/bug-hit-tables/185675/12

As well as data taken from the classic beta:
Beta test data https://github.com/magey/classic-warrior/issues/5

### CODE REVIEW

**Comment by mako-taco** [comment]
I'll upload some DPSMate screenshots of hittables after I can get a significant amount of data

File: src/game/Objects/Unit.cpp [hunk 1]
```diff
@@ -2900,8 +2912,14 @@ float Unit::MeleeSpellMissChance(Unit *pVictim, WeaponAttackType attType, int32
     // Hit chance depends from victim auras
     if (attType == RANGED_ATTACK)
         hitChance += pVictim->GetTotalAuraModifier(SPELL_AURA_MOD_ATTACKER_RANGED_HIT_CHANCE);
-    else
-        hitChance += pVictim->GetTotalAuraModifier(SPELL_AURA_MOD_ATTACKER_MELEE_HIT_CHANCE);
+    else {
+        // The first hit from gear (and likely talents) was ignored by mobs with > 10 skill difference
+        int32 hitFromAuras = pVictim->GetTotalAuraModifier(SPELL_AURA_MOD_ATTACKER_MELEE_HIT_CHANCE);
```
Comment (mako-taco) [inline src/game/Objects/Unit.cpp:2917]: Is there a way to get the total aura modifiers from gear only? The blue post seems to indicate that talents would not be considered in this penalty.

Comment (ratkosrb) [reply src/game/Objects/Unit.cpp:2917]: Both are auras, so i wouldn't think so.

**Comment by ratkosrb** [comment]
I was actually planning on porting Guybrush's implementation. His changes are more extensive, so it seems he fixed some things you forgot.

https://github.com/lh-server/core/pull/236

**Comment by mako-taco** [comment]
didnt know someone else worked on it already! awesome

**Comment by mako-taco** [comment]
I'll comment on his PR.  There are some things that I think he has implemented incorrectly.

**Pull request closed without merging**